checks/raw/shell_download_validate.go

```diff
@@ -327,6 +329,32 @@ func collectFetchPipeExecute(startLine, endLine uint, node syntax.Node, cmd, pat
 		return
 	}
 
+	var urls []*url.URL
```

Comment:
Can you create a separate function to handle the logic, maybe `isUnpinnedURLs()`?